static/app/components/searchQueryBuilder/tokens/filterKeyListBox/useFilterKeyListBox.tsx

```diff
@@ -21,45 +24,95 @@ export function useFilterKeyListBox({
   items: Array<KeyItem | KeySectionItem>;
 }) {
   const {filterKeySections} = useSearchQueryBuilder();
-  const [selectedSection, setSelectedSection] = useState<Key | null>(null);
+  const [selectedSectionKey, setSelectedSection] = useState<Key | null>(null);
 
   const sections = useMemo(() => {
-    return items.filter(itemIsSection);
+    return items.filter(item => 'options' in item);
   }, [items]);
 
   const shownItems = useMemo(
     () =>
       items.filter(item => {
         if (itemIsSection(item)) {
-          return !selectedSection || item.key === selectedSection;
+          return !selectedSectionKey || item.key === selectedSectionKey;
         }
 
         return true;
       }),
-    [items, selectedSection]
+    [items, selectedSectionKey]
   );
 
   const customMenu: CustomComboboxMenu<KeyItem | KeySectionItem> = useCallback(
     props => {
       return (
         <FilterKeyListBox
           {...props}
-          selectedSection={selectedSection}
+          selectedSection={selectedSectionKey}
           setSelectedSection={setSelectedSection}
           sections={sections}
         />
       );
     },
-    [sections, selectedSection]
+    [sections, selectedSectionKey]
   );
 
   const shouldShowExplorationMenu =
     filterValue.length === 0 && filterKeySections.length > 0;
 
+  /**
+   * Handles switching between sections with ArrowRight and ArrowLeft.
+   * Only enabled when the special menu is shown and there is a focused option.
+   * This must be implemented with onKeyDownCapture because the focused key gets
+   * reset on ArrowRight and ArrowLeft by default [1], so we must intercept the event
+   * before that happens.
+   *
+   * [1]: https://github.com/adobe/react-spectrum/blob/3691c7fc9c4f4138e268704bb776694018f04259/packages/@react-aria/combobox/src/useComboBox.ts#L171
+   */
+  const onKeyDownCapture = useCallback(
+    (
+      e: React.KeyboardEvent<HTMLInputElement>,
+      {state}: {state: ComboBoxState<KeyItem | KeySectionItem>}
+    ) => {
+      switch (e.key) {
+        case 'ArrowRight':
+        case 'ArrowLeft': {
```

Comment:
switch feels a little odd here since there's only one case. maybe its the returns at the bottom